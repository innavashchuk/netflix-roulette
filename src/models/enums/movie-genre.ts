/* eslint-disable no-shadow */
export enum MovieGenreEnum {
    drama = 'Drama',
    actionAndAdventure = 'Action & Adventure',
    comedy = 'Comedy',
    thriller = 'Thriller',
    oscarWinning = 'Ocar winning Movie',
    documentary = 'Documentary',
    horror = 'Horror',
    crime = 'Crime',
    adventure = 'Adventure',
    action = 'Action',
    fantasy = 'Fantasy',
    scienceFiction = 'Science Fiction',
    family = 'Family',
    animation = 'Animation',
    mystery = 'Mystery',
    music = 'Music',
    romance = 'Romance'
}

export const MOVIE_GENRES_LIST: Array<string> = [
    MovieGenreEnum.action,
    MovieGenreEnum.adventure,
    MovieGenreEnum.animation,
    MovieGenreEnum.comedy,
    MovieGenreEnum.crime,
    MovieGenreEnum.thriller,
    MovieGenreEnum.drama,
    MovieGenreEnum.family,
    MovieGenreEnum.fantasy,
    MovieGenreEnum.horror,
    MovieGenreEnum.music,
    MovieGenreEnum.mystery,
    MovieGenreEnum.romance,
    MovieGenreEnum.scienceFiction,
    MovieGenreEnum.documentary
];
