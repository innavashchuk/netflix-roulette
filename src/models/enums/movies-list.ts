/* eslint-disable no-shadow */
export enum SortingFieldsEnum {
  title = 'title',
  rating = 'vote_average',
  releaseDate = 'release_date'
}

export interface Sorter {
  key: string,
  name: string,
  label: string
}

export interface SorterList {
  [key: string]: Sorter
}

export enum FilterFieldEnum {
  all = 'All',
  documentary = 'Documentary',
  comedy = 'Comedy',
  horror = 'Horror',
  crime = 'Crime'
}

export interface Filter {
  key: string,
  name: FilterFieldEnum
}

export interface FilterList {
  [key: string]: Filter
}

export enum SortingDirectionEnum {
  desc = 'desc',
  asc = 'asc'
}

export const FILTER_FIELDS: FilterList = {
  [FilterFieldEnum.all]: {
    key: 'filterKey0',
    name: FilterFieldEnum.all
  },
  [FilterFieldEnum.documentary]: {
    key: 'filterKey1',
    name: FilterFieldEnum.documentary
  },
  [FilterFieldEnum.comedy]: {
    key: 'filterKey2',
    name: FilterFieldEnum.comedy
  },
  [FilterFieldEnum.horror]: {
    key: 'filterKey3',
    name: FilterFieldEnum.horror
  },
  [FilterFieldEnum.crime]: {
    key: 'filterKey4',
    name: FilterFieldEnum.crime
  }
};

export const SORTING_FIELDS: SorterList = {
  [SortingFieldsEnum.title]: {
    key: 'sortingField0',
    name: SortingFieldsEnum.title,
    label: SortingFieldsEnum.title
  },
  [SortingFieldsEnum.rating]: {
    key: 'sortingField1',
    name: SortingFieldsEnum.rating,
    label: 'rating'
  },
  [SortingFieldsEnum.releaseDate]: {
    key: 'sortingField2',
    name: SortingFieldsEnum.releaseDate,
    label: 'release date'
  }
};


export interface MoviesFilter {
  filterField: FilterFieldEnum,
  sortField: SortingFieldsEnum,
  searchValue: string,
  sortingDirection: SortingDirectionEnum 
}

