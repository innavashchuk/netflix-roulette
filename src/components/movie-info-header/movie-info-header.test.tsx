import * as React from 'react';
import {render, screen, fireEvent} from '@testing-library/react';
import MovieInfoHeader from './movie-info-header';
import {MovieGenreEnum} from '../../models/enums/movie-genre';

const MOVIE = {
    'id': 181808,
    'title': 'Star Wars: The Last Jedi',
    'tagline': 'The Saga Continues',
    'vote_average': 7.1,
    'vote_count': 4732,
    'release_date': '2017-12-13',
    'poster_path': 'https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg',
    'overview': 'Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.',
    'budget': 200000000,
    'revenue': 1325937250,
    'genres': [
        MovieGenreEnum.fantasy,
        MovieGenreEnum.adventure,
        MovieGenreEnum.scienceFiction
    ],
    'runtime': 152
};

const onSearchClick = jest.fn();

describe('MovieInfoHeader', () => {
    it('should initialize', () => {
        const {asFragment} = render(<MovieInfoHeader movie={MOVIE} onSearchClick={onSearchClick}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should call onSearchClick callback when search button is clicked', () => {
        render(<MovieInfoHeader movie={MOVIE} onSearchClick={onSearchClick}/>);
        const searchButton = screen.getByRole('button');
        expect(searchButton).toBeTruthy();
        fireEvent.click(searchButton);
        expect(onSearchClick).toHaveBeenCalled();
    });
});
