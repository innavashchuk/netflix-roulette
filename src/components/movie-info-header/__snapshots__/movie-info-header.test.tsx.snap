// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfoHeader should initialize 1`] = `
<DocumentFragment>
  <div
    class="movie-info-header"
  >
    <div
      class="movie-info-header__top"
    >
      <p
        class="logo"
      >
        <span
          class="logo__span"
        >
          netflix
        </span>
        roulette
      </p>
      <button
        class="search-button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root search-button__icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </button>
    </div>
    <div
      class="movie-info-header__container"
    >
      <img
        alt="Star Wars: The Last Jedi poster"
        class="movie-info__poster"
        src="https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg"
      />
      <div
        class="movie-info"
      >
        <div
          class="movie-info__title"
        >
          <h1
            class="movie-info__movie_title"
          >
            Star Wars: The Last Jedi
          </h1>
          <div
            class="movie-info__rating"
          >
            7.1
          </div>
        </div>
        <p>
          Adventure, Fantasy, Science Fiction
        </p>
        <div
          class="movie-info__line"
        >
          <p>
            2017
          </p>
          <p>
            152 min
          </p>
        </div>
        <p>
          Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
